src/rgw/rgw_sal_motr.cc

```diff
@@ -1378,9 +1462,31 @@ void MotrObject::gen_rand_obj_instance_name()
 {
   enum {OBJ_INSTANCE_LEN = 32};
   char buf[OBJ_INSTANCE_LEN + 1];
-
+  std::string version_id;
   gen_rand_alphanumeric_no_underscore(store->ctx(), buf, OBJ_INSTANCE_LEN);
-  key.set_instance(buf);
+  //TODO: Handle null version object case in PutObj operation.
+  // As the version ID timestamp is encoded in Base62, the maximum value
+  // for 8-characters is 62^8 - 1. This is the maximum time interval in ms.
+  constexpr uint64_t max_ts_count = 218340105584895;
+  constexpr size_t ts_len = 8, rand_id_len = 24;
```

Comment:
so 32 byte version id it will produce, which is line with AWS spec, though RGW uses/generates 31Byte version id. if we use 31 Byte here, would that be a problem.
@gregnsk @ujjwalpl @bkirunge7 shall we continue using 31Byte version id like RGW or align with what AWS spec says ?

Reply:
After more discussion, got to know version-id of 31 Bytes indeed getting created, with some additional logic on line 1487.
However better we keep intended version id length in a macro( so we can change later if needed, like to match AWS 32 byte version id) and then use it to set length for rand_id_len. Also its better to have one logic to get 31 byte or 32 byte version id based on input required length.

Reply:
FYI, the AWS spec does not require 32-byte IDs (see [definition](https://docs.aws.amazon.com/AmazonS3/latest/userguide/versioning-workflows.html)), that's just what you can observe the AWS S3 service behaving like. They could change it at any time. Min.io uses 37 bytes (UUID v4).

Reply:
We have added a macro function which will enable us to change the version-id size. Currently we are keeping version-id size as 31 bytes which is in sync with RADOS.
`#define TS_LEN 8
#define UUID_LEN 23`